### PR TITLE
[LUA] Dealer Moogle: always return table for item selection

### PR DESCRIPTION
**Pull request by jamesbradleym**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Returns a table from `getItemSelection` in `dealer_moogle.lua`
- Modifies `debugInfo` to accept items and uses the first value for `messageSpecial` which is looking for a single value, only entering this method if #items > 0
- uses the first value for `messageBasic` which is looking for a single value

## Steps to test these changes

finish event with dealer moogle

### CODE REVIEW

File: scripts/globals/dealer_moogle.lua [hunk 1]
```diff
@@ -3348,15 +3348,15 @@ local getItemSelection = function(player, list, idx, idxAlt1, idxAlt2)
...
             ---@diagnostic disable-next-line: cast-local-type
             item = itemList[list][idxAlt1][idxAlt2]
         end
 
         if list == 12 then  -- Item, Quantity
-            --- TODO: Refactor function to always return table, and find better way to determine behavior based on list type.
+            --- TODO: Find better way to determine behavior based on list type.
             ---@diagnostic disable-next-line: cast-local-type
-            item = { item } -- Tabling here to save 100 pairs of { }
+            item = item -- Tabling here to save 100 pairs of { }
```
Comment (WinterSolstice8) [inline scripts/globals/dealer_moogle.lua]: Is that comment here obsolete?

Comment (jamesbradleym) [reply scripts/globals/dealer_moogle.lua]: Ah, you beat me - yeah I just removed it

**Review by WinterSolstice8 — APPROVED** [review]
seems ok to me but will defer to Clay

**Comment by Xaver-DaRed** [comment]
This has yet to be tested

**Comment by jamesbradleym** [comment]
@Xaver-DaRed Tested with Dealer Moogle, all seems well

**Pull request merged**